components/global/Timeline/index.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { useState, useEffect, useRef } from "react";
+import Image from "next/image";
+import {
+  Container,
+  TimelineEntry,
+  TimelineExperience,
+  Type,
+  TimelineYears,
+  ExperienceItem,
+} from "./styled";
+import { monthDiff } from "../../../utils/helpers";
+
+export interface IExperience {
+  duration: {
+    start: { date?: number; month: number; year: number };
+    end?: { date?: number; month: number; year: number };
+  };
+  company: {
+    image?: string;
+    title: string;
+    website?: string;
+    industry?: string;
+    technologies?: string[];
+  };
+  meta?: {
+    width?: number;
+    left?: number;
+    level?: number;
+  };
+}
+
+export interface ITimelineEntry {
+  type: string;
+  color: string;
+  experience: IExperience[];
+  meta?: {
+    depth: number;
+  };
+}
+
+export interface ITimeline {
+  data: ITimelineEntry[];
+}
+
+interface IRangeYear {
+  start: number;
+  end: number;
+}
+
+interface TimelineProps {
+  timeline: ITimeline;
+}
+const Timeline: React.FC<TimelineProps> = ({ timeline }) => {
```

Comment:
We can merge for now, but let's pair program next week on optimizing this. This is a very important component

Reply:
Sure